src/components/inputs/number/number.component.tsx

```diff
@@ -22,12 +22,17 @@ const NumberField: React.FC<FormFieldInputProps> = ({ field, value, errors, warn
     return value ?? '';
   }, [value]);
 
+  const getNumericValue = useCallback(
+    (value: string | number) => (typeof value === 'undefined' || isNaN(Number(value)) ? undefined : Number(value)),
+    [],
+  );
+
   const handleChange = useCallback(
     (event, { value }) => {
-      const parsedValue = isEmpty(value) ? undefined : Number(value);
+      const parsedValue = getNumericValue(value);
```

Comment:
Does this alter the value on submission? for example if i don't enter any value in the number field, does the value ignored because it's undefined or is this parsed as zero?

Reply:
If there's nothing in the input field, the `value` returned by the `onChange` is `''`, hence marking it as undefined rather than `''`.